src/Amalgam/AssetManager.cpp

```diff
@@ -192,6 +175,21 @@ Entity *AssetManager::LoadEntityFromResourcePath(std::string &resource_path, std
 		delete new_entity;
 		return nullptr;
 	}
+
+	if(file_type == FILE_EXTENSION_COMPRESSED_AMALGAM_CODE)
+	{
+		new_entity->SetRoot(code, true);
+
+		EvaluableNodeReference args = EvaluableNodeReference(new_entity->evaluableNodeManager.AllocNode(ENT_ASSOC), true);
+		args->SetMappedChildNode(ENBISI_create_new_entity, new_entity->evaluableNodeManager.AllocNode(ENT_FALSE));
+		auto call_stack = Interpreter::ConvertArgsToCallStack(args, new_entity->evaluableNodeManager);
+
+		ExecutionCycleCount max_num_steps = 0, num_steps_executed = 0;
+		size_t max_num_nodes = 0, num_nodes_allocated = 0;
+		new_entity->Execute(max_num_steps, num_steps_executed, max_num_nodes, num_nodes_allocated, StringInternPool::NOT_A_STRING_ID, call_stack);
+		return new_entity;
+	}
```

Comment:
Doesn't this execute with root permissions?

Reply:
Second question: this gets executed for every loaded caml regardless if user wants it or not. How does that affect behavior? What's the before/after expectation of existing code?

Reply:
Another question here: what about all the other logic after this block? It's not clear to me that all that getting skipped is fine now...